core-api/src/main/java/com/optimizely/ab/config/audience/match/MatchTypeError.java

```diff
@@ -18,8 +18,8 @@
 package com.optimizely.ab.config.audience.match;
 
 public enum MatchTypeError {
-    UNKNOWN_MATCH_TYPE("uses an unknown match type: {}"),
-    INAPPLICABLE_CONDITION_VALUE("condition value data type is inapplicable for match type \"{}\"");
+    UNKNOWN_MATCH_TYPE("uses an unknown match type. You may need to upgrade to a newer release of the Optimizely SDK"),
+    INAPPLICABLE_CONDITION_VALUE("has an unknown condition value type. You may need to upgrade to a newer release of the Optimizely SDK");
```

Comment:
Thanks for explaining what `INAPPLICABLE_CONDITION_VALUE` is for! Can we change the message a little bit to clarify? I think this would be sufficient:
```
"has an unexpected value type. You may need to upgrade to a newer release of the Optimizely SDK"
```
After thinking about it for a while, I guess it's okay to suggest that the customer upgrade their SDK here. It could help if we eventually support non-primitive values for `exact` match, or lexicographic `gt`/`lt` semantics for strings values, etc. 🤷